src/components/Modal/src/Modal.vue

```diff
@@ -86,54 +92,35 @@ const Modal = {
 
 				const deltaY = changedTouches[0].clientY - this.touchStartFrom;
 				if (deltaY > 200) {
-					this.close();
+					this.modalApi.close();
 				} else {
 					modal.style.transform = '';
 					modal.style.overflow = '';
 				}
 			}, { once: true });
 		},
 	},
-
-	render() {
-		this.assertRootComponent();
-
-		const vnode = (
-			<div
-				class={this.$s.Modal}
-				ref="modal"
-				onTouchstart={this.onTouchstart}
-			>
-				{
-					/**
-					 * @slot modal content
-					 */
-					this.$slots.default
-				}
-			</div>
-		);
-
-		// If opened with the API modal.open()
-		if (this.$vnode.context.$vnode.componentOptions.modalFunction) {
-			return vnode;
-		}
-
-		// If opened as a component (eg. Router page)
-		this.closeRoutedModal = this.modalApi.setModalVnode(vnode);
-
-		return undefined;
-	},
 };
-
-export default Modal;
-
 </script>
 
 <style module="$s">
 .Modal {
+	width: 100%;
 	height: 100%;
-	overflow-y: auto;
 	background: #f5f6f7;
-	box-shadow: 0 0 24px 8px rgba(0, 0, 0, 0.3);
+}
+
+@media screen and (min-width: 1200px) {
+	.Modal {
```

Comment:
The modal is supposed to have 24px of padding. Right now if you try to add this padding manually to the modal, nothing visible happens on desktop. On mobile the entire modal is inset by 24px. Both aren't expected.

Do we need a ModalContent component or are we to instruct manually adding this padding selectively? The latter seems risky.

Reply:
added ModalContent component